gitlab.go

```diff
@@ -22,24 +31,44 @@ func isGitLabURL(s string) (string, bool) {
 }
 
 // findGitLabREADME tries to find the correct README filename in a repository
-func findGitLabREADME(s string) (*source, error) {
-	u, err := url.ParseRequestURI(s)
+func findGitLabREADME(repoURL string) (*source, error) {
+	user, repo, err := userAndRepoFromURL(repoURL)
+	if err != nil {
+		return nil, err
+	}
+
+	apiRepoURL := fmt.Sprintf("https://gitlab.com/api/v4/projects/%s%s%s", user, "%2F", repo)
+
+	resp, err := http.Get(apiRepoURL)
+	if err != nil {
+		return nil, err
+	}
+
+	content, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, r := range readmeNames {
-		v := u
-		v.Path += "/raw/master/" + r
+	repoInfo := gitLabRepo{}
+
+	if err := json.Unmarshal([]byte(string(content)), &repoInfo); err != nil {
```

Comment:
This is useless conversion as `ioutil.ReadAll` already returns `[]byte`.

Ref: [`ioutil.ReadAll`](https://godocs.io/io/ioutil#ReadAll)

```suggestion
	if err := json.Unmarshal(content, &repoInfo); err != nil {
```